### PR TITLE
code review fixed

### PR DESCRIPTION
**Pull request by allaVolkov**
code review fixed

### Checklist
- [x] Code compiles correctly
- [x] Relevant tests were added (unit / contract / integration)
- [ ] Relevant logs were added
- [x] Formating and linting run locally successfully
- [x] All tests passing
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

**Comment by tal-sapan** [comment]
Copying the mail with the comments (these are the remaining comments from PR #433):

**gen_test.go**
* line 40: I think this folder (testapp) shouldn't be created. The module's temp dir is under the module name, not its path.
> Fixed. As we talked adaptation of mta.yaml -> mtad.yaml also changed and tests changed (+one added)

**assembly_test.go**
* line 84: What does removeSpecialSymbols do? Why do we need it?
> To avoid line separators problem. Without this tests that worked in Windows failed in Linux.
Maybe we can sometime understand the root of the problem and remove this…

**manifest_test.go**
* line 215: Change the test name: `should skip the module when it has no path` (I think that's enough details since the test suite is about the module having build-artifact-name)
> done

**meta_test.go**
* lines 35-36: Use createTmpFile here
> Done

* line 73: Use createTmpFile here
> Done

* lines 133-134: This can be removed. It's done inside createMtahtml5Folder.
> Done

* line 202: I think this folder (testapp) shouldn't be created. The module's temp dir is under the module name, not its path.
> Done

line 214 / function createTmpFile:
* The function createFileInGivenPath in module_arch_test.go does the same (and createFile creates it in a test folder) so one of these functions can be removed
* About the function name: the created file isn't temporary. It's just empty (created then closed).
* (In whatever function remains) There should be a check for errors when creating the file and when closing it. Since Close will fail if Create failed, the error check for Create should be before calling Close.
> Done.  New set of functions defined: 
`createDirInTempFolder`
`createFileInTmpFolder`
`createFileInGivenPath`
`getFullPathInTmpFolder`
used in most of places in code   

**internal/artifacts/testdata/assembly/mtad.yaml**
* line 37: Can one of the last 3 modules (m8, m9, m10) be inside a subfolder? I don't think we have a test like this for assemble
> Done. m9

File: internal/artifacts/testdata/mtahtml5/mta.yaml [hunk 1]
```diff
@@ -12,6 +12,8 @@ modules:
    requires:
     - name: uaa_mtahtml5
     - name: dest_mtahtml5
+   build-parameters:
```
Comment (tal-sapan) [inline internal/artifacts/testdata/mtahtml5/mta.yaml]: Why did you add this?

File: internal/artifacts/artifacts_suite_test.go [hunk 2]
```diff
@@ -44,3 +45,28 @@ func getFileContent(filePath string) string {
 	expected, _ := ioutil.ReadFile(filePath)
 	return removeSpecialSymbols(expected)
 }
+
+func createFileInTmpFolder(projectName string, path ...string) {
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/artifacts_suite_test.go]: I discussed this situation with Shimi. His solution is to copy such utils for unit tests. Please address him if we need to change the approach.

File: cmd/commands_suite_test.go [hunk 3]
```diff
@@ -46,3 +47,21 @@ func executeAndProvideOutput(execute func()) string {
...
 }
+
+func createFileInTmpFolder(projectName string, path ...string) {
+	file, err := os.Create(getFullPathInTmpFolder(projectName, path...))
+	Ω(err).Should(Succeed())
+	err = file.Close()
+	Ω(err).Should(Succeed())
+}
+
+func createDirInTempFolder(projectName string, path ...string) {
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply cmd/commands_suite_test.go]: renamed

File: internal/artifacts/artifacts_suite_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/artifacts_suite_test.go]: done

File: internal/artifacts/mtad.go [hunk 4]
```diff
@@ -164,19 +163,11 @@ func removeBuildParamsFromMta(loc dir.ITargetPath, mtaStr *mta.MTA) error {
...
-		// if path not exists, use the module name
-		if err != nil && os.IsNotExist(err) {
-			modulePath = filepath.Join(loc.GetTargetTmpDir(), module.Name)
-			_, e := os.Stat(modulePath)
-			if e != nil {
-				return err
-			}
-			module.Path = module.Name
-		}
+	modulePath := filepath.Join(loc.GetTargetTmpDir(), module.Name)
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/mtad.go]: According to Natalia for development descriptor it has to be not empty. If it's empty (according to validation implementation) we assume path == module name. If her answer to the mail I sent will confirm it then this code is correct and we have to adjust pack.

File: internal/artifacts/meta_test.go [hunk 5]
```diff
@@ -43,8 +42,8 @@ var _ = Describe("Meta", func() {
 			})).Should(HaveOccurred())
 		})
 		It("Wrong platform", func() {
-			dir.CreateDirIfNotExist(getTestPath("result", ".mtahtml5_mta_build_tmp", "ui5app2"))
-			dir.CreateDirIfNotExist(getTestPath("result", ".mtahtml5_mta_build_tmp", "testapp"))
+			createDirInTempFolder("mtahtml5", "ui5app2")
+			createDirInTempFolder("mtahtml5", "testapp")
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/meta_test.go]: error checks added

File: internal/artifacts/meta_test.go [hunk 6]
```diff
@@ -130,13 +129,24 @@ cli_version:["x"]
...
 			createMtahtml5TmpFolder()
-			file, _ := os.Create(getTestPath("result", ".mtahtml5_mta_build_tmp", "xs-security.json"))
-			file.Close()
 			ep := dir.Loc{SourcePath: getTestPath("mtahtml5"), TargetPath: getResultPath()}
 			Ω(generateMeta(&ep, &ep, &ep, &ep, false, "cf")).Should(Succeed())
-			Ω(readFileContent(&dir.Loc{SourcePath: getTestPath("result", ".mtahtml5_mta_build_tmp", "META-INF"), Descriptor: "dep"})).
+			Ω(readFileContent(&dir.Loc{SourcePath: getFullPathInTmpFolder("mtahtml5", "META-INF"), Descriptor: "dep"})).
 				Should(Equal(readFileContent(&dir.Loc{SourcePath: getTestPath("golden"), Descriptor: "dep"})))
 		})
+		It("Generate Mtad - fails on missing paths in temp folder", func() {
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/meta_test.go]: description of the test and the message changed

File: internal/artifacts/meta_test.go [hunk 7]
```diff
@@ -130,13 +129,24 @@ cli_version:["x"]
...
+			createFileInTmpFolder("mtahtml5", "xs-security.json")
+			ep := dir.Loc{SourcePath: getTestPath("mtahtml5"), TargetPath: getResultPath()}
+			mtaStr, err := ep.ParseFile()
+			Ω(err).Should(Succeed())
+			for i := range mtaStr.Modules {
+				mtaStr.Modules[i].Path = ""
+			}
+			err = genMetaInfo(&ep, &ep, &ep, false, "neo", mtaStr)
+			Ω(err).Should(HaveOccurred())
+			Ω(err.Error()).Should(ContainSubstring(`failed to clean the build parameters from the "ui5app" module`))
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/meta_test.go]: message changed

File: internal/artifacts/testdata/mtahtml5/mta.yaml (hunk 1 above)
Comment (allaVolkov) [reply internal/artifacts/testdata/mtahtml5/mta.yaml]: in order to pass successfully setManifest if called with unsupported platform. To have an option to succeed with wrong paths till mtad generation. Maybe trick is wrong and we have to remove unsupported (by platform) modules from mtad...

File: internal/artifacts/mtad.go [hunk 8]
```diff
@@ -14,7 +15,11 @@ import (
 	"github.com/SAP/cloud-mta-build-tool/internal/archive"
 	"github.com/SAP/cloud-mta-build-tool/internal/buildops"
 	"github.com/SAP/cloud-mta-build-tool/internal/logs"
-	"strings"
+)
+
+const (
+	invalidPlatformMsg = `the invalid target platform "%s"; supported platforms are: "cf", "neo", "xsa"`
```
Comment (tal-sapan) [inline internal/artifacts/mtad.go]: I think the `the` at the beginning should be removed

File: internal/artifacts/testdata/mtahtml5/mta.yaml (hunk 1 above)
Comment (tal-sapan) [reply internal/artifacts/testdata/mtahtml5/mta.yaml]: Sorry, I don't understand... will a test fail if you remove it? If so, which test and why? And why is it different from before these fixes?

File: internal/artifacts/mtad.go (hunk 8 above)
Comment (allaVolkov) [reply internal/artifacts/mtad.go]: Julia worked on this. Let's leave for Paola

File: internal/artifacts/testdata/mtahtml5/mta.yaml (hunk 1 above)
Comment (tal-sapan) [reply internal/artifacts/testdata/mtahtml5/mta.yaml]: After discussion - the test sends neo instead of cf, so the metadata generation ignores the empty module paths, and the mtad generation fails on them
